azurelinuxagent/ga/monitor.py

```diff
@@ -173,118 +238,48 @@ def stop(self):
             self.join()
 
     def join(self):
-        self.event_thread.join()
+        self.monitor_thread.join()
 
     def stopped(self):
         return not self.should_run
 
-    def init_protocols(self):
-        # The initialization of ProtocolUtil for the Monitor thread should be done within the thread itself rather
-        # than initializing it in the ExtHandler thread. This is done to avoid any concurrency issues as each
-        # thread would now have its own ProtocolUtil object as per the SingletonPerThread model.
-        self.protocol_util = get_protocol_util()
-        self.protocol = self.protocol_util.get_protocol()
-        self.health_service = HealthService(self.protocol.get_endpoint())
-
-    def init_imds_client(self):
-        wireserver_endpoint = self.protocol_util.get_wireserver_endpoint()
-        self.imds_client = get_imds_client(wireserver_endpoint)
-
     def is_alive(self):
-        return self.event_thread is not None and self.event_thread.is_alive()
+        return self.monitor_thread is not None and self.monitor_thread.is_alive()
 
     def start(self):
-        self.event_thread = threading.Thread(target=self.daemon)
-        self.event_thread.setDaemon(True)
-        self.event_thread.setName(self.get_thread_name())
-        self.event_thread.start()
+        self.monitor_thread = threading.Thread(target=self.daemon)
```

Comment:
Moving all the periodic operations to their own classes made the monitor thread very simple